From 893c23824bc98f361f4d30d6baa3cbe636652dc3 Mon Sep 17 00:00:00 2001
From: Doug Ledford <dledford@redhat.com>
Date: Wed, 24 Sep 2014 17:33:06 -0400
Subject: [infiniband] qib: Cleanup qib_register_observer()

Message-id: <798e20c263574e571c25bec942687cea2c3d35a2.1411579814.git.dledford@redhat.com>
Patchwork-id: 96701
O-Subject: [Patch RHEL7 13/19] IB/qib: Cleanup qib_register_observer()
Bugzilla: 1085961
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

From: Dan Carpenter <dan.carpenter@oracle.com>

Returning directly is easier to read than do-nothing gotos.  Remove the
duplicative check on "olp" and pull the code in one indent level.

Signed-off-by: Dan Carpenter <dan.carpenter@oracle.com>
Acked-by: Mike Marciniszyn <mike.marciniszyn@intel.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit 186f8ba062f796221d51077342f3ba5202838e9f)
Signed-off-by: Doug Ledford <dledford@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/qib/qib_diag.c b/drivers/infiniband/hw/qib/qib_diag.c
index 07f9030..5dfda4c 100644
--- a/drivers/infiniband/hw/qib/qib_diag.c
+++ b/drivers/infiniband/hw/qib/qib_diag.c
@@ -694,28 +694,23 @@ int qib_register_observer(struct qib_devdata *dd,
      const struct diag_observer *op)
 {
  struct diag_observer_list_elt *olp;
- int ret = -EINVAL;
+ unsigned long flags;
 
  if (!dd || !op)
-  goto bail;
- ret = -ENOMEM;
+  return -EINVAL;
  olp = vmalloc(sizeof *olp);
  if (!olp) {
   pr_err("vmalloc for observer failed\n");
-  goto bail;
+  return -ENOMEM;
  }
- if (olp) {
-  unsigned long flags;
 
-  spin_lock_irqsave(&dd->qib_diag_trans_lock, flags);
-  olp->op = op;
-  olp->next = dd->diag_observer_list;
-  dd->diag_observer_list = olp;
-  spin_unlock_irqrestore(&dd->qib_diag_trans_lock, flags);
-  ret = 0;
- }
-bail:
- return ret;
+ spin_lock_irqsave(&dd->qib_diag_trans_lock, flags);
+ olp->op = op;
+ olp->next = dd->diag_observer_list;
+ dd->diag_observer_list = olp;
+ spin_unlock_irqrestore(&dd->qib_diag_trans_lock, flags);
+
+ return 0;
 }
 
 /* Remove all registered observers when device is closed */
-- 
1.7.1